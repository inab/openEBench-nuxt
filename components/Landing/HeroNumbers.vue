<template>
	<v-parallax :height="containerHeight" :src="material2">
		<v-container>
			<v-row align="center" justify="center" class="root">
				<v-col align="center" cols="12" sm="3">
					<div class="counter-item">
						<span class="icon ion-leaf" />
						<div class="text">
							<h4 class="text-h3">
								<animated-number :value="communityCount" />
							</h4>
							<h6 class="title">
								{{ 'Community' | pluralize(communityCount) }}
							</h6>
						</div>
					</div>
				</v-col>
				<v-col align="center" cols="12" sm="3">
					<div class="counter-item">
						<span class="icon ion-ios-lightbulb-outline" />
						<div class="text">
							<h4 class="text-h3">
								<animated-number :value="toolsCount" />
							</h4>
							<h6 class="title">
								{{ 'Tool' | pluralize(toolsCount) }}
							</h6>
						</div>
					</div>
				</v-col>
				<v-col align="center" cols="12" sm="3">
					<div class="counter-item">
						<span class="icon ion-ios-world-outline" />
						<div class="text">
							<h4 class="text-h3 font-weight-normal">
								<animated-number :value="resourcesCount" />
							</h4>
							<h6 class="title">
								{{ 'Resource' | pluralize(resourcesCount) }}
							</h6>
						</div>
					</div>
				</v-col>
				<v-col align="center" cols="12" sm="3">
					<div class="counter-item">
						<span class="icon ion-ios-world-outline" />
						<div class="text">
							<h4 class="text-h3 font-weight-normal">
								<animated-number :value="projectsCount" />
							</h4>
							<h6 class="title">
								{{ 'Project Space' | pluralize(projectsCount) }}
							</h6>
						</div>
					</div>
				</v-col>
			</v-row>
		</v-container>
	</v-parallax>
</template>

<script>
import AnimatedNumber from '~/components/Landing/AnimatedNumber';
import material2 from '~/static/images/material2_parallax.webp';

export default {
	name: 'HeroNumbers',
	components: {
		AnimatedNumber,
	},
	props: {
		communityCount: {
			type: Number,
			required: true,
		},
		toolsCount: {
			type: Number,
			required: true,
		},
		resourcesCount: {
			type: Number,
			required: true,
		},
		projectsCount: {
			type: Number,
			required: true,
		},
	},
	data() {
		return {
			material2,
		};
	},
	computed: {
		containerHeight() {
			return this.$vuetify.breakpoint.xs ? 450 : 170;
		},
	},
};
</script>
