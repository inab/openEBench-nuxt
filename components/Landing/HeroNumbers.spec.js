import { shallowMount } from '@vue/test-utils';
import HeroNumbers from './HeroNumbers.vue';

const factory = (propsData) => {
	return shallowMount(HeroNumbers, {
		...createComponentMocks({}),
		mocks: {
			$vuetify: { breakpoint: {} },
		},
		propsData,
	});
};

const propsData = {
	toolsCount: 10,
	communityCount: 5,
	resourcesCount: 20,
};

describe('HeroNumbers', () => {
	it('is instantiated', () => {
		const wrapper = factory(propsData);
		expect(wrapper).toBeTruthy();
	});

	it('should match snapshot', () => {
		const wrapper = factory(propsData);
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('should have a container height of 350 on small devices', async () => {
		const wrapper = factory(propsData);

		wrapper.vm.$vuetify.breakpoint.xs = true;
		await wrapper.vm.$nextTick();
		expect(wrapper.vm.containerHeight).toBe(450);
	});

	it('should have a container height of 170 on bigger devices', async () => {
		const wrapper = factory(propsData);

		wrapper.vm.$vuetify.breakpoint.mdAndUp = true;
		await wrapper.vm.$nextTick();
		expect(wrapper.vm.containerHeight).toBe(170);
	});
});
