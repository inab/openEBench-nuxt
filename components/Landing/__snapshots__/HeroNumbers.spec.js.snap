// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroNumbers should match snapshot 1`] = `
"<v-parallax-stub height=\\"170\\" alt=\\"\\" src=\\"\\">
  <v-container-stub tag=\\"div\\">
    <v-row-stub tag=\\"div\\" align=\\"center\\" justify=\\"center\\" class=\\"root\\">
      <v-col-stub cols=\\"12\\" sm=\\"3\\" tag=\\"div\\" align=\\"center\\">
        <div class=\\"counter-item\\"><span class=\\"icon ion-leaf\\"></span>
          <div class=\\"text\\">
            <h4 class=\\"text-h3\\">
              <animated-number-stub value=\\"5\\"></animated-number-stub>
            </h4>
            <h6 class=\\"title\\">
              Community
            </h6>
          </div>
        </div>
      </v-col-stub>
      <v-col-stub cols=\\"12\\" sm=\\"3\\" tag=\\"div\\" align=\\"center\\">
        <div class=\\"counter-item\\"><span class=\\"icon ion-ios-lightbulb-outline\\"></span>
          <div class=\\"text\\">
            <h4 class=\\"text-h3\\">
              <animated-number-stub value=\\"10\\"></animated-number-stub>
            </h4>
            <h6 class=\\"title\\">
              Tool
            </h6>
          </div>
        </div>
      </v-col-stub>
      <v-col-stub cols=\\"12\\" sm=\\"3\\" tag=\\"div\\" align=\\"center\\">
        <div class=\\"counter-item\\"><span class=\\"icon ion-ios-world-outline\\"></span>
          <div class=\\"text\\">
            <h4 class=\\"text-h3 font-weight-normal\\">
              <animated-number-stub value=\\"20\\"></animated-number-stub>
            </h4>
            <h6 class=\\"title\\">
              Resource
            </h6>
          </div>
        </div>
      </v-col-stub>
      <v-col-stub cols=\\"12\\" sm=\\"3\\" tag=\\"div\\" align=\\"center\\">
        <div class=\\"counter-item\\"><span class=\\"icon ion-ios-world-outline\\"></span>
          <div class=\\"text\\">
            <h4 class=\\"text-h3 font-weight-normal\\">
              <animated-number-stub value=\\"0\\"></animated-number-stub>
            </h4>
            <h6 class=\\"title\\">
              Project Space
            </h6>
          </div>
        </div>
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</v-parallax-stub>"
`;
